File: CozyHouse/Dominio/Entities/Apartment.cs
namespace SistemaAPI.Entidades;

public class Apartment {
    public int Id { get; set; }
    public string Door { get; set; } = string.Empty;
    public int Floor { get; set; }
    public double Price { get; set; }
    public double Area { get; set; }
    public int NumberOfRooms { get; set; }
    public int NumberOfBathrooms { get; set; }
    public bool IsAvailable { get; set; }

    // FK y navegación al edificio
    public int BuildingId { get; set; }
    public Building? Building { get; set; }

    // Relación con el alquiler
    public Rental? Rental { get; set; }
}

File: CozyFront/src/app/models/apartment.ts
export interface Apartment {
    id: number,
    code: string,
    door: string,
    floor: number,
    price: number,
    area: number,
    numberOfRooms: number,
    numberOfBathrooms: number,
    buildingId: number,
    hasLift: boolean,
    hasGarage: boolean,
    isAvailable: boolean,
    imageUrls?: string[],
    districtName: string,
    streetName: string
}
